### PR TITLE
Build: Fix exceptions

### PR DESCRIPTION
**Pull request by stsewd**
We were passing the path to the constructor,
but that argument is actually the message.

The message is already defined in the exception itself.

Fixes https://github.com/readthedocs/readthedocs.org/issues/10538

### CODE REVIEW

File: readthedocs/core/utils/filesystem.py [hunk 1]
```diff
@@ -89,17 +89,21 @@ def safe_open(
...
             log.info("Path traversal via symlink.")
-            raise SymlinkOutsideBasePath(path)
+            raise SymlinkOutsideBasePath()
 
     _assert_path_is_inside_docroot(resolved_path)
 
+    # The encoding is valid only if the file opened is a text file,
+    # this functions is used to read both types of files (text and binary),
+    # so we can't specify the encoding here.
+    # pylint: disable=unspecified-encoding
```
Comment (humitos) [inline readthedocs/core/utils/filesystem.py:106]:
```suggestion
    # The encoding is valid only if the file opened is a text file,
    # this function is used to read both types of files (text and binary),
    # so we can't specify the encoding here.
    # pylint: disable=unspecified-encoding
```

**Review by humitos — APPROVED** [review]
> We were passing the path to the constructor, but that argument is actually the message.

What about checking the `message=` type is string in the initializer?

https://github.com/readthedocs/readthedocs.org/blob/bbb200ba2e42d032843e3f8da170c61b76af514e/readthedocs/doc_builder/exceptions.py#L9-L19

Or, cast it into string?

I'm thinking about a way to avoid this issue since we've hit long time ago and it was hard to track. I remember talking to Anthony about this and we weren't able to figure  it out what was happening.

**Comment by stsewd** [comment]
> What about checking the message= type is string in the initializer? Or, cast it into string?

Do you mean raising an exception if the type is incorrect? I'm +1 on that instead of casting, since looks like most of the time, passing that argument to exceptions that have an already defined default message is a mistake.

**Comment by stsewd** [comment]
Didn't find any other exceptions where we were overriding the default message like this.

**Pull request merged**